llama-index-integrations/embeddings/llama-index-embeddings-voyageai/llama_index/embeddings/voyageai/base.py

```diff
@@ -22,79 +19,78 @@ class VoyageEmbedding(BaseEmbedding):
             You can either specify the key here or store it as an environment variable.
     """
 
-    _model: Any = PrivateAttr()
+    client: Optional[voyageai.Client]
 
     def __init__(
         self,
-        model_name: str = "voyage-01",
+        model_name: str = "voyage-2",
         voyage_api_key: Optional[str] = None,
-        embed_batch_size: int = DEFAULT_VOYAGE_BATCH_SIZE,
+        embed_batch_size: Optional[int] = None,
         callback_manager: Optional[CallbackManager] = None,
         **kwargs: Any,
     ):
-        if voyage_api_key:
-            voyageai.api_key = voyage_api_key
-        self._model = voyageai
-
+        if embed_batch_size is None:
+            embed_batch_size = 72 if model_name in ["voyage-2", "voyage-02"] else 7
         super().__init__(
             model_name=model_name,
             embed_batch_size=embed_batch_size,
             callback_manager=callback_manager,
             **kwargs,
         )
+        self.client = voyageai.Client(api_key=voyage_api_key)
 
     @classmethod
     def class_name(cls) -> str:
         return "VoyageEmbedding"
 
     def _get_query_embedding(self, query: str) -> List[float]:
         """Get query embedding."""
-        return self._model.get_embedding(
-            query, model=self.model_name, input_type="query"
-        )
+        return self.client.embed(
+            [query], model=self.model_name, input_type="query"
+        ).embeddings
```

Comment:
I suppose it should be `embeddings[0]` here

Reply:
It could be, but in that case, the type of `embeddings` is wrong.

Reply:
https://github.com/voyage-ai/voyageai-python/blob/main/voyageai/embeddings_object.py

<img width="821" alt="Screenshot 2024-02-27 at 11 13 10" src="https://github.com/voyage-ai/llama_index/assets/160063452/b30dae13-0852-4eb7-a1da-f449df9620b2">

embeddings seems to be Optional[List[float]]